prover/rust-toolchain

```diff
@@ -1 +1 @@
-nightly-2023-08-21
+nightly-2024-02-01
```

Comment:
Could having different toolchains for core and prover be an issue? Not saying it necessarily is, but looks a bit sus.